### ISSUE TITLE
8.1.2 → 8.2.0 (- 9.0.0) regression: `softThreshold: true` gets ignored.

### ISSUE THREAD

**Issue by stroobandt** [highcharts/highcharts#15061]
#### Expected behaviour
The secondary axis should cross the X-axis at a high Y value (e.g. 2000) due to `softThreshold: true` set in the `plotOptions` of `area` graphs.

#### Actual behaviour
The secondary axis crosses the X-axis at zero Y value.

#### Live demo with steps to reproduce

- [Expected behaviour of highstock 8.1.2](https://jsfiddle.net/stroobandt/togpwkmy/)
- [Actual behaviour of highstock 8.2.0](https://jsfiddle.net/stroobandt/4pL8hc2d/)

#### Product version

Highstock 8.2.0 (to 9.0.0)
From what I read in the forum and in other bug reports, most probably also Highcharts 8.2.0 (to 9.0.0).

#### Affected browser(s)

|||
|:-|:-|
|Vivaldi | 3.6.2165.34 (Stable channel) (64-bit)|
|Revision | 240ce41835957243f3c94b0b857c0f4344bcecd3|
|OS | Linux|
|JavaScript | V8 8.8.278.14|
|User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.99 Safari/537.36|

**Comment by KacperMadej** [comment]
Hi @stroobandt 

Thank you for reporting the problem.

Setting `threshold: null` resolves the problem - https://jsfiddle.net/BlackLabel/k5pqsyr0/

Please let me know if this fixes the problem and if the issue could be closed as resolved.

**Comment by stroobandt** [comment]
I agree, in your jsfiddle it resolves the issue. However, in my more involved use case the problem remains.
I am only having difficulty to reproduce it in a simple enough jsfiddle. I will keep trying.
I cannot share the actual use case because it involves sensitive patient data.

A [documentation issue](https://api.highcharts.com/highcharts/plotOptions.series.softThreshold) remains anyhow. `threshold` is nowhere mentioned in the `softThreshold` entry.
Furthermore, I do not understand the need for this additional `threshold` property.

**Comment by stroobandt** [comment]
As I said, this bug may be closed provided the documentation receives a little more attention.

**Comment by KacperMadej** [comment]
The behavior is correct as 0 is the default `threshold` for `area` and ["If a number is given, the Y axis will scale to the threshold."](https://api.highcharts.com/highcharts/series.area.threshold).

The difference between 8.1.2 and 8.2.0+ behavior is caused by https://github.com/highcharts/highcharts/issues/13538

I am not how docs could be improved significantly to provide a clear interpretation of the option. Maybe mentioning `threshold` option like below would be enough?

["If `softThreshold` is `true`, the Y axis starts at 0."](https://api.highcharts.com/highcharts/series.area.softThreshold) + ` (or threshold option).`.

**Comment by stroobandt** [comment]
There are a couple of things going on here. First of all, the intended behaviour could be stated a bit more explicitly, such as:

> #### `softThreshold: boolean`
> When this is `true`, the series will not cause the Y axis to cross the zero plane (or a different level set by `threshold` option), unless the data actually crosses the plane.

However, the issue I ran into can be summarised as follows:

1. The `threshold` base level was originally left untouched, hence defaulting to zero.
2. `softThreshold` was set to `true`.
3. Nonetheless, the Y axis of [my area graph](https://jsfiddle.net/stroobandt/zk0x23vg/) was still crossing the zero plane. This is not according to the `softThreshold` spec.
4. As by your suggestion, I had to _additionally_ define `threshold: null` to obtain the desired behaviour of `softThreshold: true`. This necessity is nowhere mentioned in the current documentation of `softThreshold: boolean`.

In conclusion:

- Either `softThreshold: true` is no longer doing what it is supposed to do, independently of the `threshold` level (this is my take)
- or, the specification of `softThreshold` has crept to something else, which was not properly documented.

**Comment by KacperMadej** [comment]
`softThreshold` were not working as documented before 8.2.0 and this was fixed in #13538

> Nonetheless, the Y axis of my area graph was still crossing the zero plane. 

Zero (threshold) is not crossed. The yAxis for area type series is on the right side and the minimum is 0.

**Comment by stroobandt** [comment]
How can the minimum be zero when the `data` of the area graph are all large values?

        data: [2144.0, 2176.0, 2135.6, 2148.5, 2216.4, 2194.1, 2095.6, 2054.4, 2029.9, 2071.5, 2106.4, 2129.2],

The area is a visual presentation aspect, that has nothing to do with the Y-axis values.
The definition of the `softThreshold` property refers to values; not to pixels.
I beg to differ from your analysis of this issue.

**Comment by KacperMadej** [comment]
> How can the minimum be zero when the data of the area graph are all large values?

The y axis, which is connected to the area series, has a minimum equal to the area series threshold setting - zero. [`If a number is given, the Y axis will scale to the threshold.`](https://api.highcharts.com/highcharts/series.area.threshold) - it is given, it is zero.

> The area is a visual presentation aspect, that has nothing to do with the Y-axis values.

A area type series shows a 2D zone/area limited vertically by y values (data) and the series threshold. At least this is by default and by design - it is possible to change this e.g. by changing `threshold` option to `null`.

> The definition of the softThreshold property refers to values; not to pixels.

Yes, fully agree. I must have missed something regarding your remark here - please elaborate.

> I beg to differ from your analysis of this issue.

The issue is still labeled as `pending reply` until the issue will be clear. I am sorry that this takes so long to clear up - I appreciate your patinece.
